tests/cpp/utils.h

```diff
@@ -169,6 +169,22 @@ class PredicatedChecker : public kir::IrVisitor {
     return isPredicated(tv->name(), kernel);
   }
 
+  static bool isPredicatedByIfThenElse(
+      StmtNameType tv_name,
+      kir::Kernel* kernel) {
+    PredicatedChecker checker(tv_name, kernel->topLevelExprs());
+    return checker.predicated_ite_;
+  }
+  static bool isSharedMemoryPredicatedByIfThenElse(kir::Kernel* kernel) {
+    for (auto tv : kernel->allTvs()) {
+      if (tv->getMemoryType() == MemoryType::Shared &&
```

Comment:
Should we only check smem operands? smem epilogue does not use cp.async, so it should not have this limitation.

Reply:
changed to `If a tv is CpAsync copied to shared memory, then LdMatrix copied to registers, it is used in mma and should use inline predicate instead of if-then-else`